packages/patternfly-4/react-inline-edit-extension/src/components/InlineEdit/editableTableBody.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { TableContext, TableBody, isRowExpanded } from '@patternfly/react-table';
+
+import PropTypes from 'prop-types';
+import { TableEditConfirmation } from './constants';
+import { showIdWarnings } from './utils/utils';
+
+const propTypes = {
+  ...TableBody.propTypes,
+  editConfig: PropTypes.shape({
+    editConfirmationType: PropTypes.oneOf(Object.values(TableEditConfirmation)),
+    onEditCellClicked: PropTypes.func,
+    onEditConfirmed: PropTypes.func,
+    onEditCanceled: PropTypes.func
+  }).isRequired,
+  /** Function that is fired when user clicks on a row if not editing.  */
+  onRowClick: PropTypes.func
+};
+
+const defaultProps = {
+  ...TableBody.defaultProps,
+  editConfig: null,
+  onRowClick: () => undefined
+};
+
+const resolveCascadeEditability = rows => {
+  const isRowExpandedIndexes = new Set(
+    rows.map((row, idx) => (isRowExpanded(row, rows) ? idx : null)).filter(row => row !== null)
+  );
+
+  // flag parents and their children which are edited together
+  rows
+    .filter(
+      (row, idx) =>
+        row.parent !== undefined &&
+        row.isEditing &&
+        isRowExpandedIndexes.has(idx) &&
+        row.isEditableTogetherWithParent &&
+        rows[row.parent].isEditing
+    )
+    .forEach(row => {
+      rows[row.parent].isChildEditing = true;
+      row.isParentEditing = true;
+    });
+
+  const lastVisibleRow = rows.filter((row, idx) => !row.parent || isRowExpandedIndexes.has(idx)).pop();
+
+  // flag last parent row if there are only descendants under it
+  if (lastVisibleRow && lastVisibleRow.isParentEditing) {
+    let parentRow = lastVisibleRow;
+    while (parentRow.parent !== undefined && parentRow.isEditableTogetherWithParent) {
+      parentRow = rows[parentRow.parent];
+    }
+    parentRow.isLastVisibleParent = true;
+  }
+};
+
+const onRow = (event, row, rowProps, computedData, { onRowClick, editConfig }) => {
+  const { target } = event;
+  const cell = target.closest('[data-key]');
+  const cellNumber = parseInt(cell && cell.getAttribute('data-key'));
+  const hasCellNumber = !Number.isNaN(cellNumber);
+
+  let onEditCellClicked;
+
+  if (hasCellNumber && editConfig && typeof editConfig.onEditCellClicked === 'function') {
+    // resolve closest (e.g. for dropdowns) usable id of a clicked element inside a cell
+    const idElement = target.closest('[id]');
+    const elementId = idElement && cell.contains(idElement) ? idElement.getAttribute('id') || null : null;
+
+    if (!elementId) {
+      showIdWarnings(row, target);
+    }
+
+    onEditCellClicked = () => {
+      editConfig.onEditCellClicked(event, row, {
+        ...rowProps,
+        columnIndex: cellNumber,
+        elementId
+      });
+    };
+  }
+
+  // give priority to fire onChange/onBlur callbacks
+
+  setTimeout(() => {
```

Comment:
I decided to stop checking if the onEdit callback should be fired. Consequently, I renamed `onEditCellChanged` to  `onEditCellClicked`. The checking is up to the consumer and the usage is more permissive.